### PR TITLE
delete deprecated BAT_* parameters

### PR DESCRIPTION
**Pull request by dagar**
QGC has been updated, so we should be good to delete these for the next release. Do not merge until verified.

Saves ~ 1.25 kB of flash.

### CODE REVIEW

File: src/drivers/batt_smbus/batt_smbus.cpp [hunk 1]
```diff
@@ -57,9 +57,8 @@ BATT_SMBUS::BATT_SMBUS(I2CSPIBusOption bus_option, const int bus, SMBus *interfa
 	int battsource = 1;
 	int batt_device_type = (int)SMBUS_DEVICE_TYPE::UNDEFINED;
 
-	param_set(param_find("BAT_SOURCE"), &battsource);
-	param_get(param_find("BAT_SMBUS_MODEL"), &batt_device_type);
-
+	param_set(param_find("BAT1_SOURCE1"), &battsource);
```
Comment (BazookaJoe1900) [inline src/drivers/batt_smbus/batt_smbus.cpp:60]: need to be BAT1_SOURCE
is there chance not to use param_find()? and use 'px4::params::BAT1_SOURCE'?

**Review by BazookaJoe1900 — COMMENTED** [review]
small note about parameter name

File: src/drivers/batt_smbus/batt_smbus.cpp (hunk 1 above)
Comment (dagar) [reply src/drivers/batt_smbus/batt_smbus.cpp:60]: Moving to `px4::params::BAT1_SOURCE'` would be cleaner right now, but might be a (small) step in the wrong direction if there's any interest in supporting multiple instances of batt_smbus.

Comment (BazookaJoe1900) [reply src/drivers/batt_smbus/batt_smbus.cpp:60]: and what about "BAT1_SOURCE1" -> "BAT1_SOURCE"

Comment (LorenzMeier) [reply src/drivers/batt_smbus/batt_smbus.cpp]: Fixed.

**Comment by dagar** [comment]
I'll bring this in post v1.12.

**Comment by dagar** [comment]
Rebased on master, but there were quite a few conflicts.

**Pull request merged**